tools/toolchains/gcc.py

```diff
@@ -166,7 +166,11 @@ def get_dep_option(self, object):
         return ["-MD", "-MF", dep_path]
 
     def get_compile_options(self, defines, includes):
-        return ['-D%s' % d for d in defines] + ['@%s' % self.get_inc_file(includes)]
+        opts = ['-D%s' % d for d in defines] + ['@%s' % self.get_inc_file(includes)]
```

Comment:
```['@%s' % self.get_inc_file(includes)]``` asks gcc to read the file as more command line options.

```['-imacros %s' % self.get_inc_file(includes)]``` asks gcc to read the file and only define macros.

```['-include %s' % self.get_inc_file(includes)]``` asks gcc to read the file and include everything.

I think you want one of the last two